src/tests/gc-e2e/GuestConfiguration.Tests.ps1

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# GuestConfiguration.Tests.ps1
+param (
+    [Parameter(Mandatory)]
+    [string] $PolicyPackage
+)
+
+Describe 'Validate GuestConfiguration NRP' {
+    # Perform monitoring - Get
+    Context "Get" {
+        BeforeAll {
+            $result = Get-GuestConfigurationPackageComplianceStatus -Path $PolicyPackage -Verbose
+        }
+
+        It 'Correct resource count - 20' {    
+            $result | Should -Not -BeNullOrEmpty
+            $result.resources | Should -HaveCount 20
+        }
+    }
+
+    # Perform Remediation - Set
+    Context "Set" {
+        BeforeAll {
+            $result = Start-GuestConfigurationPackageRemediation -Path $PolicyPackage -Verbose
+        }
+
+        It 'Ensure resource count = 20' {    
```

Comment:
>Ensure resource count = 20 [](http://example.com/codeflow?start=12&length=26)

These are not resources. There is a single resource, the Universal NRP. They are resource instances.

Change the name to 'Ensure the total resource instances count' and skip the '= 20' part (we will change this number soon)